LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -848,7 +854,7 @@ extension GlucoseStore {
             self.cacheStore.managedObjectContext.performAndWait {
                 for sample in samples {
                     let object = CachedGlucoseObject(context: self.cacheStore.managedObjectContext)
-                    object.create(from: sample, provenanceIdentifier: self.provenanceIdentifier)
+                    object.create(from: sample, provenanceIdentifier: self.provenanceIdentifier, healthKitStorageDelay: self.healthKitStorageDelay)
```

Comment:
This should use the same `self.sharingDenied ? nil : self.healthKitStorageDelay` logic. 

Also, consider a helper property that wraps the above logic.